apps/src/common.rs

```diff
@@ -313,6 +313,26 @@ pub fn priority_from_query_string(url: &url::Url) -> Option<Priority> {
     }
 }
 
+fn send_h3_dgram(
+    conn: &mut quiche::Connection, flow_id: u64, dgram_content: &[u8],
+) -> quiche::Result<()> {
+    info!(
+        "sending HTTP/3 DATAGRAM on flow_id={} with data {:?}",
+        flow_id, dgram_content
+    );
+
+    let len = octets::varint_len(flow_id) + dgram_content.len();
+    let mut d = vec![0; len];
```

Comment:
I know this is similar to current implementation, but reallocating the entire datagram and copying it is extremely wasteful

Reply:
The API should expose a method that allows to pass a generic that implements AsRef<[u8]>, and then those can be queued with the flow_id and the actual copying only happens when they are sent on the wire. At the very list a method that accepts a Vec<u8> and avoids reallocating